src/query/functions-v2/src/scalars/control.rs

```diff
@@ -173,4 +186,108 @@ pub fn register(registry: &mut FunctionRegistry) {
             ValueRef::Scalar(Some(_)) => Ok(Value::Scalar(true)),
         },
     );
+
+    fn cal_in<T: ArgType>(
+        lhs: ValueRef<T>,
+        rhs: ValueRef<ArrayType<T>>,
+    ) -> Result<Value<BooleanType>, String>
+    where
+        T::Scalar: HashtableKeyable,
+    {
+        // must be scalar in rhs, if rhs contains column, it'll be rewrited to `a = b1 or a = b2 or a = b3` in type_checker
+        let array = rhs.as_scalar().unwrap();
+        let mut set = StackHashSet::<_, 128>::with_capacity(T::column_len(array));
```

Comment:
We use a global hashset in v1. Can we achieve this in v2?

Reply:
Yes, it's not in high-priority, we plan to do it after new expression is done